### PR TITLE
chore: Clean up `parser` comments

### PR DESCRIPTION
**Pull request by liuxingbaoyu**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | `Fixes #1, Fixes #2` 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

1. Many JSDOC comments are broken or inconsistent with the real code, I delete those useless type declarations
2. Fixed some internal typos
3. Deleted several useless TODOs

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/15252"></a>

### CODE REVIEW

File: packages/babel-parser/src/parser/comments.ts [hunk 1]
```diff
@@ -7,32 +7,20 @@ import type { Undone } from "./node";
...
- *                                         with minimal size (|end - start|)
  */
 export type CommentWhitespace = {
-  start: number;
-  end: number;
-  comments: Array<Comment>;
-  leadingNode: Node | null;
-  trailingNode: Node | null;
-  containingNode: Node | null;
+  start: number; // the start of the whitespace token.
```
Comment (nicolo-ribaudo) [inline packages/babel-parser/src/parser/comments.ts:12]: Nit, but if we use jsdoc-like comments editors will show the as the description in autocompletion:
```js
  /**
   * The start of the whitespace token.
   */
  start: number;
```

even if probably comments for start/end/comments are not needed.

File: packages/babel-parser/src/parser/comments.ts [hunk 2]
```diff
@@ -237,16 +206,15 @@ export default class CommentsParser extends BaseParser {
    * method later. In this case the identifier is not part of the AST and we
    * should sync the knowledge to commentStacks
    *
-   * For example, when parsing */
-  // async /* 1 */ function f() {}
-  /*
+   * ↓↓↓ We had escaped "\/" to "\\/" for comments
+   * For example, when parsing *\/
+   * // async /* 1 *\/ function f() {}
```
Comment (nicolo-ribaudo) [inline packages/babel-parser/src/parser/comments.ts:211]: I actually prefer the old comment style, so that it didn't need escapes :stuck_out_tongue:

Comment (JLHwung) [reply packages/babel-parser/src/parser/comments.ts:211]: Can you try if https://github.com/JLHwung/babel/commit/6eee6d0be2bff890a1f0a7ad6384d56f6aecf551 works? I added U+200B between `*` and `/`.

Comment (liuxingbaoyu) [reply packages/babel-parser/src/parser/comments.ts:211]: Wow this is awesome, thank you!

File: packages/babel-parser/src/parser/comments.ts (hunk 1 above)
Comment (nicolo-ribaudo) [reply packages/babel-parser/src/parser/comments.ts:12]: It should work when overing over start:

image

**Review by JLHwung — APPROVED** [review]

**Review by nicolo-ribaudo — APPROVED** [review]

**Pull request merged**